### PR TITLE
feat(components): [badge] add show-zero prop

### PR DESCRIPTION
**Pull request by iSolitudinis**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/badge/src/badge.ts [hunk 1]
```diff
@@ -24,6 +24,13 @@ export const badgeProps = buildProps({
    * @description hidden badge.
    */
   hidden: Boolean,
+  /**
+   * @description Whether to display the badge, even if provided value equals 0..
+   */
+  showZero: {
```
Comment (btea) [inline packages/components/badge/src/badge.ts:30]:
```suggestion
  showZero: Boolean
```

File: packages/components/badge/src/badge.vue [hunk 2]
```diff
@@ -39,6 +39,8 @@ const content = computed<string>(() => {
   return `${props.value}`
 })
 
+const isZero = computed(() => !props.showZero && Number(content.value) <= 0)
```
Comment (btea) [inline packages/components/badge/src/badge.vue:42]:
```suggestion
const isZero = computed(() => !props.showZero && Number(content.value) === 0)
```

File: packages/components/badge/src/badge.vue [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <slot />
     <transition :name="`${ns.namespace.value}-zoom-in-center`">
       <sup
-        v-show="!hidden && (content || isDot)"
+        v-show="!hidden && (!isZero || isDot)"
```
Comment (btea) [inline packages/components/badge/src/badge.vue:6]:
```suggestion
        v-show="showSup"
```
```ts
const showSup = computed(() => {
  if (props.hidden) {
    return false
  }
  if (!props.showZero) {
    return Number(content.value) !== 0
  }
  return props.isDot
})
```

**Comment by jw-foss** [comment]
Got to ask under what scenario we would want the badge to draw attention.

**Comment by jw-foss** [comment]
Try [this](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYmFkZ2UgdmFsdWU9XCIwXCIgY2xhc3M9XCJpdGVtXCI+XG4gICAgPGVsLWJ1dHRvbj5jb21tZW50czwvZWwtYnV0dG9uPlxuICA8L2VsLWJhZGdlPlxuICA8ZWwtYmFkZ2UgOnZhbHVlPVwiMTAwXCIgOm1heD1cIjEwXCIgY2xhc3M9XCJpdGVtXCI+XG4gICAgPGVsLWJ1dHRvbj5yZXBsaWVzPC9lbC1idXR0b24+XG4gIDwvZWwtYmFkZ2U+XG48L3RlbXBsYXRlPlxuXG48c3R5bGUgc2NvcGVkPlxuLml0ZW0ge1xuICBtYXJnaW4tdG9wOiAxMHB4O1xuICBtYXJnaW4tcmlnaHQ6IDQwcHg7XG59XG48L3N0eWxlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319) out, you don't really need this props to show zero.

CleanShot 2022-12-16 at 23 54 38@2x


Simply pass a string instead of a number to `value`. I think this can be closed since the problem you were trying to solve no longer exists. Feel free to reopen this.

**Comment by VanillaZen** [comment]
> Got to ask under what scenario we would want the badge to draw attention.

Do not want to display when the value is 0

**Pull request closed without merging**